nightshift/datastore_transactions.py

```diff
@@ -0,0 +1,44 @@
+# -*- coding: utf-8 -*-
+
+from nightshift.datastore import Resource
+
+
+def insert_or_ignore(session, model, **kwargs):
+    """
+    Adds a new record to given table (model) or ignores if the same.
+
+    Args:
+        session:                sqlalchemy.Session instance
+        model:                  one of datastore table classes
+        kwargs:                 new record values as dictionary
+
+    Returns:
+        instance of inserted or duplicate record
+    """
+    instance = session.query(model).filter_by(**kwargs).first()
+    if not instance:
+        instance = model(**kwargs)
+        session.add(instance)
+    return instance
```

Comment:
Should this return some information as to whether the record was created or not? Not sure the context in which this will be used but I can imagine knowing if a record already exists in the db might prove useful. (This can be learned by checking if the returned record has a `nid` so this isn't strictly necessary but not sure how you feel about relying on that.

Reply:
Thanks for the suggestion. I agree, the current approach is not consistent. I like your general idea to return an instance if something has been changed and nothing otherwise. Will try to apply the same rule in other methods in this module.